main.go

```diff
@@ -159,13 +169,41 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Add readiness probe
+	//	Add readiness probe
 	if err := mgr.AddReadyzCheck("readyz", storagecluster.ReadinessChecker); err != nil {
-		setupLog.Error(err, "unable add a readiness check")
+		setupLog.Error(err, "unable to add a readiness check")
 		os.Exit(1)
 	}
-	storagecluster.ReadinessSet()
 
+	// apiclient.New() returns a client without cache.
+	// cache is not initialized before mgr.Start()
+	// we need this because we need to interact with OperatorCondition
+	apiClient, err := apiclient.New(mgr.GetConfig(), apiclient.Options{
+		Scheme: mgr.GetScheme(),
+	})
+
+	// Set OperatorCondition Upgradeable to True
+	//
+	// We have to at least default the condition to True or
+	// OLM will use the Readiness condition via our readiness probe instead:
+	// https://olm.operatorframework.io/docs/advanced-tasks/communicating-operator-conditions-to-olm/#setting-defaults
+
+	// TODO: handle error
```

Comment:
I will definitely require this before merging. If we can't set the condition you should immediately exit and let the Pod crash loop.

Reply:
updated